### PR TITLE
Refactor: Advance file transfer and change the way files are named

### PR DESCRIPTION
**Pull request by jiantaolovebingbing**

### CODE REVIEW

File: src/app/containers/Workspace/Workspace.jsx [hunk 1]
```diff
@@ -374,9 +387,8 @@ const mapStateToProps = (state) => {
         secondaryWidgets
     };
 };
-
-
 const mapDispatchToProps = (dispatch) => ({
+    addGcodeFile: (name, uploadName, size, lastModifiedDate, img) => dispatch(workspaceActions.addGcodeFile(name, uploadName, size, lastModifiedDate, img)),
```
Comment (zjiefee) [inline src/app/containers/Workspace/Workspace.jsx]: param is fileinfo

File: src/app/widgets/PrintingMaterial/Material.jsx [hunk 2]
```diff
@@ -95,6 +95,7 @@ class Material extends PureComponent {
             this.setState({
                 newName: event.target.value
             });
+            console.log(this.stat.newName);
```
Comment (zjiefee) [inline src/app/widgets/PrintingMaterial/Material.jsx]: console need remove

File: src/app/widgets/PrintingOutput/Output.jsx [hunk 3]
```diff
@@ -54,18 +55,21 @@ class Output extends PureComponent {
                 });
                 return;
             }
-
-            const gcodePath = this.props.gcodePath;
+            const { originalName, uploadName } = this.props.gcodeFileName;
```
Comment (zjiefee) [inline src/app/widgets/PrintingOutput/Output.jsx]: why gcodeFileName?

File: src/app/widgets/PrintingOutput/Output.jsx [hunk 4]
```diff
@@ -54,18 +55,21 @@ class Output extends PureComponent {
...
             document.location.href = '/#/workspace';
             window.scrollTo(0, 0);
-            const filename = path.basename(gcodePath);
-            jQuery.get(gcodePath, (result) => {
+            let filename = path.basename(originalName);
+            if (path.extname(filename) && ['.stl', '.obj'].includes(path.extname(filename).toLowerCase())) {
```
Comment (zjiefee) [inline src/app/widgets/PrintingOutput/Output.jsx]: filename 后缀处理放到一个共同的地方吧

**Comment by parachvte** [comment]
Pls change your PR title to match contributing guidelines.

**Pull request merged**